nav2_amcl/src/amcl_node.cpp

```diff
@@ -494,6 +494,16 @@ AmclNode::globalLocalizationCallback(
   pf_init_ = false;
 }
 
+void
+AmclNode::initialGuessCallback(
+  const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+  const std::shared_ptr<nav2_msgs::srv::SetInitialPose::Request> req,
+  std::shared_ptr<nav2_msgs::srv::SetInitialPose::Response>/*res*/)
+{
+  std::lock_guard<std::recursive_mutex> cfl(mutex_);
+  initialPoseReceived(std::make_shared<geometry_msgs::msg::PoseWithCovarianceStamped>(req->pose));
```

Comment:
`initialPoseReceived` locks the mutex, no need to do that here

Reply:
Removed unnecessary mutex lock.